Dockerfile

```diff
@@ -23,5 +8,36 @@ RUN mix local.hex --force && \
 
 ENV MIX_ENV=prod
 
-ADD . .
-RUN elixir --erl "-smp enable" /usr/local/bin/mix do deps.get --only prod, phx.swagger.generate, compile, phx.digest, distillery.release --verbose
+ADD apps apps
+ADD config config
+ADD mix.* /root/
+
+RUN mix do deps.get --only prod, phx.swagger.generate, compile, phx.digest
+
+ADD rel/ rel/
+
+RUN mix distillery.release --verbose
+
+# The one the elixir image was built with
+FROM debian:stretch
```

Comment:
I think [`elixir:1.9.1`](https://github.com/c0b/docker-elixir/commit/8c92148d07fc89a57706dcb22c926a96e73a4194#diff-51c95e8cd4f39fd0a0428238d2574e74) pulls from just `erlang:22` (in contrast to what used to be in this file, `erlang:22.0`), which nowadays is built on [`debian:buster`](https://github.com/erlang/docker-erlang-otp/blob/636231974186645b6c96c6c888e61fd086394c3a/22/Dockerfile).

I'm not sure what happens if the two build stages are inconsistent. I vaguely remember copying over `stretch` when building Arrow before getting a crash and realizing it had changed.

Reply:
thanks, good catch!

Reply:
Actually, it looks like this is Stretch (and the builds are failing with Buster).

- `elixir:1.9.1` was built against `erlang:22.1.3`: https://hub.docker.com/layers/elixir/library/elixir/1.9.1/images/sha256-ab9c60fe6cdafc15fead89cbf99c2950aaec6fb94d8b0694d867eca372e6dcc3
- `erlang:22.1.3` was built against `buildpack-deps:stretch`: https://github.com/erlang/docker-erlang-otp/commit/b5f9e48145117d414c1475216a3e0e3bc550c664#diff-b6d767d2f8ed5d21a44b0e5886680cb9

Reply:
Ah, interesting. So I guess even though the dockerfile at the time simply specified `erlang 22` (as far as I can tell from the GitHub history), it has been forever snapshotted on the docker hub at `22.1.3`?

How are you getting to `erlang:22.1.3` from that link, btw? I see the ENV var which has that `OTP_VERSION` but I'm not seeing a jump to a corresponding docker image. I'm assuming it has to do with the `ADD file` at the top, but I'm coming up short trying to trace that hash.

Reply:
I assumed it based on the OTP_VERSION: I couldn't find a way back from that hash.

Reply:
Also based on the builds crashing when I tried to use `buster` :)

Reply:
Haha, fair enough! Just wondering if I was missing something.